File: IAmThereApi/RequestModels/UserRegisterModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace IAmThereApi.RequestModels
{
    public class UserRegisterModel
    {
        [Required]
        [DataType(DataType.EmailAddress)]
        [StringLength(100, MinimumLength = 1)]
        public string Email { get; set; }
        [Required]
        [StringLength(255, MinimumLength = 8)]
        public string Password { get; set; }
        [Required]
        [StringLength(60, MinimumLength = 1)]
        public string FirstName { get; set; }
        [Required]
        [StringLength(60,MinimumLength = 1)]
        public string LastName { get; set; }
    }
}

File: iamthere/lib/models/userRegisterModel.dart
class UserRegisterModel {
  final String _email;
  final String _password;
  final String _firstName;
  final String _lastName;

  UserRegisterModel(
      this._email, this._password, this._firstName, this._lastName);

  Map<String, dynamic> toJson() => {
        'email': _email,
        'password': _password,
        'firstName': _firstName,
        'lastName': _lastName,
      };
}
